### PR TITLE
Use AppSyncRealTimeClient Pod

### PR DESCRIPTION
**Pull request by lawmicha**
The consolidation work can be found here: https://github.com/awslabs/aws-mobile-appsync-sdk-ios/pull/338

The code was then moved over and checked in at https://github.com/aws-amplify/aws-appsync-realtime-client-ios

pod has been published as 1.0.0 to represent the current state of things without major bug fix or changes, except the call outs would be

- Interceptors are currently implemented by the consumer
- A subscription factory class is also implemenented by the consumer.
- both possible set of functionality that needs extra effort to be pulled into the shared package
- A lot of places required making classes public

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by lawmicha** [comment]
datastore unit test failing on pod install

```
[!] Unable to find a specification for `AppSyncSubscriptionClient (~> 0.0.1)` depended upon by `AmplifyPlugins/AWSAPIPlugin`
```

File: AmplifyPlugins/API/AWSAPICategoryPlugin/Operation/AWSGraphQLSubscriptionOperation.swift [hunk 1]
```diff
@@ -93,14 +93,26 @@ final public class AWSGraphQLSubscriptionOperation<R: Decodable>: GraphQLSubscri
...
+                                                             eventHandler: { [weak self] event, _ in
+            self?.onAsyncSubscriptionEvent2(event: event)
         })
 
     }
 
+    private func onAsyncSubscriptionEvent2(event: SubscriptionItemEvent) {
+        switch event {
+        case .connection(let subscriptionConnectionEvent):
+            onSubscriptionEvent2(subscriptionConnectionEvent)
```
Comment (royjit) [inline AmplifyPlugins/API/AWSAPICategoryPlugin/Operation/AWSGraphQLSubscriptionOperation.swift]: Same '2' here..

File: AmplifyPlugins/API/AWSAPICategoryPlugin/Operation/AWSGraphQLSubscriptionOperation.swift [hunk 2]
```diff
@@ -93,14 +93,26 @@ final public class AWSGraphQLSubscriptionOperation<R: Decodable>: GraphQLSubscri
         }
 
         // Create subscription
+
         subscriptionItem = subscriptionConnection?.subscribe(requestString: request.document,
                                                              variables: request.variables,
-                                                             onEvent: { [weak self] event in
-            self?.onAsyncSubscriptionEvent(event: event)
+                                                             eventHandler: { [weak self] event, _ in
+            self?.onAsyncSubscriptionEvent2(event: event)
```
[the remark this answers is not in the document]
Comment (lawmicha) [reply AmplifyPlugins/API/AWSAPICategoryPlugin/Operation/AWSGraphQLSubscriptionOperation.swift]: woops, refactoring response of SubscriptionItem's event handler, renaming and removing old handler

File: AmplifyPlugins/API/AWSAPICategoryPlugin/Operation/AWSGraphQLSubscriptionOperation.swift (hunk 1 above)
Comment (lawmicha) [reply AmplifyPlugins/API/AWSAPICategoryPlugin/Operation/AWSGraphQLSubscriptionOperation.swift]: renamed and removed old handler

**Pull request merged**